operator/src/main/java/org/bf2/operator/managers/MetricsManager.java

```diff
@@ -0,0 +1,111 @@
+import org.bf2.operator.operands.KafkaCluster;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import java.util.stream.IntStream;
+
+import static org.bf2.operator.managers.InformerManager.filter;
+
+@Startup
+@ApplicationScoped
+public class MetricsManager {
+    static final String KAFKA_INSTANCE_BROKERS_DESIRED_COUNT = "kafka_instance_brokers_desired_count";
+    static final String KAFKA_INSTANCE_PARTITION_LIMIT = "kafka_instance_partition_limit";
+    static final String KAFKA_INSTANCE_MAX_MESSAGE_SIZE_LIMIT = "kafka_instance_max_message_size_limit";
+    static final String KAFKA_INSTANCE_CONNECTION_LIMIT = "kafka_instance_connection_limit";
+    static final String KAFKA_INSTANCE_CONNECTION_CREATION_RATE_LIMIT = "kafka_instance_connection_creation_rate_limit";
+    static final Tag OWNER = Tag.of("owner", "KafkaInstanceMetricsManager");
+
+    @Inject
+    ResourceInformerFactory resourceInformerFactory;
+
+    @Inject
+    MeterRegistry meterRegistry;
+
+    @Inject
+    KubernetesClient kubernetesClient;
+
+    private volatile ResourceInformer<Kafka> kafkaResourceInformer;
+
+    protected synchronized void createInformer() {
+        if (kafkaResourceInformer != null) {
+            return;
+        }
+        kafkaResourceInformer = resourceInformerFactory.create(Kafka.class, filter(kubernetesClient.resources(Kafka.class, KafkaList.class)), new ResourceEventHandler<>() {
+            @Override
+            public void onAdd(Kafka kafka) {
+                createMetrics(kafka);
+            }
+
+            @Override
+            public void onUpdate(Kafka oldObj, Kafka newObj) {
+                createMetrics(newObj);
+            }
+
+            @Override
+            public void onDelete(Kafka obj, boolean deletedFinalStateUnknown) {
+                Search.in(meterRegistry).tags(buildKafkaInstanceTags(obj)).meters().forEach(meterRegistry::remove);
+
+            }
+
+            private void createMetrics(Kafka kafka) {
+                Tags tags = buildKafkaInstanceTags(kafka);
+                meterRegistry.gauge(KAFKA_INSTANCE_BROKERS_DESIRED_COUNT, tags, kafka, this::replicas);
+                meterRegistry.gauge(KAFKA_INSTANCE_PARTITION_LIMIT, tags, kafka, k -> kafkaConfigValue(k, KafkaCluster.MAX_PARTITIONS));
+                meterRegistry.gauge(KAFKA_INSTANCE_MAX_MESSAGE_SIZE_LIMIT, tags, kafka, k -> kafkaConfigValue(k, KafkaCluster.MESSAGE_MAX_BYTES));
+
+                if (kafka.getSpec() != null && kafka.getSpec().getKafka() != null) {
+                    IntStream.range(0, kafka.getSpec().getKafka().getReplicas()).forEach(ordinal -> {
+                        Tags brokerTags = Tags.concat(tags, Tags.of(Tag.of("broker", String.valueOf(ordinal))));
+
+                        if (kafka.getSpec().getKafka().getListeners() != null) {
+                            kafka.getSpec().getKafka().getListeners().stream().filter(l -> "external".equals(l.getName())).forEach(l -> {
```

Comment:
Consider a constant for `"external"` since it's shared between this and `AbstractKafkaCluster`.